### PR TITLE
Update to Cadence v1.0.0-preview.39

### PR DESCRIPTION
**Pull request by turbolent**
## Description

Automatically update to:
- [onflow/cadence v1.0.0-preview.39](https://github.com/onflow/cadence/releases/tag/v1.0.0-preview.39)
- [onflow/flow-go-sdk v1.0.0-preview.42](https://github.com/onflow/flow-go-sdk/releases/tag/v1.0.0-preview.42)
- [onflow/flow-go 14f9fddeda2b](https://github.com/onflow/flow-go/commit/14f9fddeda2b)
- [onflow/flow-emulator v1.0.0-preview.36](https://github.com/onflow/flow-emulator/releases/tag/v1.0.0-preview.36)

## Summary by CodeRabbit

- **Dependency Updates**
  - Updated multiple dependencies to their latest versions for improved performance and security.

- **Improvements**
  - Improved logging mechanism for clearer log management.
  - Simplified handling of transaction hashes and payload fields for better clarity and efficiency.
  - Streamlined logic for processing transaction event attributes, enhancing maintainability.
  - Enhanced type safety in transaction building process by refining variable types.
  - Added block hash information to transaction receipts for better data context.
  - Updated test cases to reflect changes in blockchain state and improve coverage of transaction retrieval.

### CODE REVIEW

**Comment by turbolent** [comment]
There are several changes in flow-go and geth which need to be resolved:
- flow-go's transaction event payload no longer includes the block hash, but it is needed for the receipt
- flow-go's event constructors take fewer arguments
- Logging in geth seems to have changed significantly

**Comment by turbolent** [comment]
@m-Peter @sideninja The fixes in the dependencies are merged now:
- https://github.com/onflow/flow-go/pull/6249
- https://github.com/onflow/flow-go/pull/6251
- https://github.com/onflow/flow-emulator/pull/717

Releases have been tagged in the dependencies, and I updated everything again, which resulted in #382, which I merged into this PR.

It looks like the EVM implementation changed once again and block transaction hashes have been replaced by a transaction hash root. However, I don't know anything about these. From what I can tell this should be the last issue that has to be addressed. Let me know how I can assist!

cc @ramtinms

File: models/transaction.go [hunk 1]
```diff
@@ -188,30 +188,24 @@ func (tc TransactionCall) MarshalBinary() ([]byte, error) {
...
-		ContractAddress:  common.HexToAddress(txEvent.ContractAddress),
-		GasUsed:          txEvent.GasConsumed,
-		TransactionIndex: uint(txEvent.Index),
-		BlockHash:        common.HexToHash(txEvent.BlockHash),
+		BlockNumber:       big.NewInt(int64(txEvent.BlockHeight)),
+		Type:              txEvent.TransactionType,
+		TxHash:            txEvent.Hash,
+		ContractAddress:   common.HexToAddress(txEvent.ContractAddress),
+		GasUsed:           txEvent.GasConsumed,
+		CumulativeGasUsed: txEvent.GasConsumed, // todo use cumulative after added to the tx result
```
Comment (turbolent) [inline models/transaction.go:204]: This might be a bad merge (re-)introducing it, not sure

**Comment by m-Peter** [comment]
@turbolent We have incrementally fixed all the breaking changes, via:
- https://github.com/onflow/flow-evm-gateway/pull/383
- https://github.com/onflow/flow-evm-gateway/pull/381

So now we have `github.com/onflow/cadence v1.0.0-preview.39` in https://github.com/onflow/flow-evm-gateway/pull/381, which a :heavy_check_mark:  CI :pray:

**Comment by sideninja** [comment]
Closing this one in favour of https://github.com/onflow/flow-evm-gateway/pull/381
There are a lot of breaking changes that needs addressing.

**Comment by turbolent** [comment]
Awesome, great work @m-Peter and @sideninja! 👏

**Pull request closed without merging**